src/viam/app/data_client.py

```diff
@@ -145,6 +148,69 @@ def __eq__(self, other: object) -> bool:
                 return str(self) == str(other)
             return False
 
+    @dataclass
+    class TabularDataPoint:
+        """Represents a tabular data point and its associated metadata."""
+
+        part_id: str
+        """The robot part ID"""
+
+        resource_name: str
+        """The resource name"""
+
+        resource_subtype: str
+        """The resource subtype. Ex: `rdk:component:sensor`"""
+
+        method_name: str
+        """The method used for data capture. Ex" `Readings`"""
+
+        time_captured: datetime
+        """The time at which the data point was captured"""
+
+        organization_id: str
+        """The organization ID"""
+
+        location_id: str
+        """The location ID"""
+
+        robot_name: str
+        """The robot name"""
+
+        robot_id: str
+        """The robot ID"""
+
+        part_name: str
+        """The robot part name"""
+
+        method_parameters: Mapping[str, Any]
+        """Additional parameters associated with the data capture method"""
+
+        tags: List[str]
+        """A list of tags associated with the data point"""
+
+        payload: Mapping[str, Any]
```

Comment:
Because this payload is coming from a `proto.Struct`, you might want to make this `Mapping[str, ValueTypes]` (`from viam.utils import ValueTypes`)